src/test/java/seedu/address/model/internship/InternshipTest.java

```diff
@@ -0,0 +1,112 @@
+package seedu.address.model.internship;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_APPLICATIONSTATUS_OPTIVER;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_COMPANY_NAME_OPTIVER;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_DURATION_OPTIVER;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_REQUIREMENT_OPTIVER;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_ROLE_OPTIVER;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_STARTDATE_OPTIVER;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalInternships.JANESTREET;
+import static seedu.address.testutil.TypicalInternships.OPTIVER;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.testutil.InternshipBuilder;
+
+class InternshipTest {
+
+    @Test
+    public void asObservableList_modifyList_throwsUnsupportedOperationException() {
+        Internship internship = new InternshipBuilder().build();
+        assertThrows(UnsupportedOperationException.class, () -> internship.getRequirements().remove(0));
+    }
+
+    @Test
+    public void isSameInternship() {
+        // same object -> returns true
+        assertTrue(JANESTREET.isSameInternship(JANESTREET));
+
+        // null -> returns false
+        assertFalse(JANESTREET.isSameInternship(null));
+
+        // same companyName and same role, all other attributes different -> returns true
+        Internship editedJaneStreet = new InternshipBuilder(JANESTREET)
+                        .withApplicationStatus(VALID_APPLICATIONSTATUS_OPTIVER)
+                        .withStartDate(VALID_STARTDATE_OPTIVER)
+                        .withDuration(VALID_DURATION_OPTIVER).withRequirements(VALID_REQUIREMENT_OPTIVER)
+                        .build();
+        assertTrue(JANESTREET.isSameInternship(editedJaneStreet));
+
+        // different companyName, all other attributes same -> returns false
+        editedJaneStreet = new InternshipBuilder(JANESTREET).withCompanyName(VALID_COMPANY_NAME_OPTIVER).build();
+        assertFalse(JANESTREET.isSameInternship(editedJaneStreet));
+
+        // companyName differs in case, all other attributes same -> returns false
+        Internship editedBob = new InternshipBuilder(OPTIVER)
+                .withCompanyName(VALID_COMPANY_NAME_OPTIVER.toLowerCase()).build();
```

Comment:
I believe you should rename this from editedBob. This appears multiple times throughout this PR